### PR TITLE
[DROOLS-6508] Implement Aggregation

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @kostola @jiripetrlik

See https://issues.redhat.com/browse/DROOLS-6508

To be viewed/merged after #3740

### CODE REVIEW

File: kie-pmml-trusty/kie-pmml-commons/src/main/java/org/kie/pmml/commons/model/expressions/KiePMMLAggregate.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    @Override
+    public Object evaluate(final ProcessingDTO processingDTO) {
+        Optional<Object> fieldValue =  getFromPossibleSources(name, processingDTO);
+        if (!fieldValue.isPresent()) {
+            return null;
+        }
+        if (!(fieldValue.get() instanceof String)) {
+            throw new IllegalArgumentException("Can not evaluate aggregate on field " + name + " whose value is " + fieldValue.get());
+        }
+        String[] stringFieldValues = ((String) fieldValue.get()).split(" ");
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-commons/src/main/java/org/kie/pmml/commons/model/expressions/KiePMMLAggregate.java:59]: Are you sure this split is safe? For example, isn't possible to have a value like `fld1    fld2`?

File: kie-pmml-trusty/kie-pmml-commons/src/main/java/org/kie/pmml/commons/model/expressions/KiePMMLAggregate.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        String[] groupBy = null;
+        if (groupField != null) {
+            Optional<Object> groupValue = getFromPossibleSources(name, processingDTO);
+            if (groupValue.isPresent()) {
+                if (!(groupValue.get() instanceof String)) {
+                    throw new IllegalArgumentException("Can not evaluate aggregate on groupField " + groupField + " whose value is " + groupValue.get());
+                } else {
+                    groupBy = ((String) groupValue.get()).split(" ");
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-commons/src/main/java/org/kie/pmml/commons/model/expressions/KiePMMLAggregate.java:67]: Same here

File: kie-pmml-trusty/kie-pmml-commons/src/main/java/org/kie/pmml/commons/model/expressions/KiePMMLAggregate.java (hunk 1 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-commons/src/main/java/org/kie/pmml/commons/model/expressions/KiePMMLAggregate.java:59]: @danielezonca @jiripetrlik 
Reading again [here](http://dmg.org/pmml/v4-4-1/Transformations.html#xsdElement_Aggregate) my opinion is that - as for Lag, Aggregate is meant to work with set of records. I'm going to disable it.

File: kie-pmml-trusty/kie-pmml-commons/src/main/java/org/kie/pmml/commons/model/expressions/KiePMMLAggregate.java (hunk 2 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-commons/src/main/java/org/kie/pmml/commons/model/expressions/KiePMMLAggregate.java:67]: See above

**Comment by gitgabrio** [comment]
Closed because currently not possible to implement (it requires set of records - i.e. multiple inputs - currently unmanaged)

**Pull request closed without merging**